### PR TITLE
Ruby: Avoid recursive extraction into tokens

### PR DESCRIPTION
**Pull request by hvitved**
For some reason, tokens may have reserved word children. This PR avoid recursing into such children, which means we will not have overlapping tokens.

### CODE REVIEW

File: ruby/ql/test/library-tests/ast/TreeSitter.expected [hunk 1]
```diff
@@ -2041,14 +2037,10 @@ control/cases.rb:
 #   57|           1: [ReservedWord] :
 #   57|         3: [ReservedWord] ,
 #   57|         4: [HashSplatNil] **nil
-#   57|           0: [ReservedWord] **
```
Comment (aibaars) [inline ruby/ql/test/library-tests/ast/TreeSitter.expected:2044]: This doesn't look right, `**` and `nil` are separate tokens. It is possible to write things like

```
def foo(** 
    # comment
   nil) end
```
In this case you'd still end up with overlapping `comment` and `hashsplatnil` tokens.

File: ruby/ql/test/library-tests/ast/erb/Erb.expected [hunk 2]
```diff
@@ -50,7 +50,6 @@ erbAstNodes
 | template.html.erb:31:5:31:9 |  end  |
 | template.html.erb:31:10:31:11 | %> |
 | template.html.erb:31:12:33:5 | \n</ul>\n<%%>\n |
-| template.html.erb:33:1:33:3 | <%% |
```
Comment (aibaars) [inline ruby/ql/test/library-tests/ast/erb/Erb.expected:53]: This token does not seem to overlap with any other one, so we probably shouldn't drop it.

**Review by aibaars — COMMENTED** [review]
This does not seem to be the right approach.

File: ruby/ql/test/library-tests/ast/TreeSitter.expected (hunk 1 above)
Comment (hvitved) [reply ruby/ql/test/library-tests/ast/TreeSitter.expected:2044]: I wonder why `HashSplatNil` is a `Token` in the first place then, when actually it is defined as the sequence of two tokens?

**Comment by hvitved** [comment]
Closing this one, as it would be better to still include leaf tokens (but not have tokens with children, like e.g. `HashSplatNil`.

**Pull request closed without merging**